setup.py

```diff
@@ -25,6 +27,108 @@
+       Python. Tools like debhelper pass the --install-layout=deb option to
+       the setup script while building a Debian package so that its installs
+       files into /usr/ not /usr/local/.
+
+    '''
+    if([x for x in sys.path if 'dist-package' in x]):
+        return(True)
+    else:
+        return(False)
+
+
+def check_conf_file_location():
+    '''Retrieve data_files values for setuptools.setup()
+
+    :returns: data_files value that can be used by setuptools.setup()
+              like the following example.
+              [('etc/', ['redfish-client/etc/redfish-client.conf']),
+               ('share/redfish-client/templates',
+                ['redfish-client/templates/manager_info.template',
+                 'redfish-client/templates/bla.template'])]
+    :rtype: structure compatible with setuptools.setup()
+    '''
+    file_location = []
+    # Add conf file according to prefix
+    if(distutils.sysconfig.PREFIX == '/usr' and not isdebianbased()):
+        file_location.append(('/etc/',
+                              getfiles('redfish-client/etc')))
+    else:
+        file_location.append(('etc/',
+                              getfiles('redfish-client/etc')))
+    # Add template files
+    file_location.append(('share/redfish-client/templates',
+                          getfiles('redfish-client/templates')))
+    return(file_location)
+
+# Install software
 setuptools.setup(
     setup_requires=['pbr'],
-    pbr=True)
+    pbr=True,
+    data_files=check_conf_file_location())
+
+# Define conf file location
+configfile_location = ''
+templatedir_location = ''
+for file_location in check_conf_file_location():
+    if('etc' in file_location[1][0]):
+        if(distutils.sysconfig.PREFIX == '/usr' and isdebianbased()):
+            configfile_location = os.path.join(
+                '/usr/local',
+                file_location[0],
+                os.path.basename(file_location[1][0]))
+        elif('/etc/' in file_location[0]):
+            configfile_location = os.path.join(
+                file_location[0],
+                os.path.basename(file_location[1][0]))
+        else:
+            configfile_location = os.path.join(
+                distutils.sysconfig.PREFIX,
+                file_location[0],
+                os.path.basename(file_location[1][0]))
+    elif('templates' in file_location[1][0]):
+        if(distutils.sysconfig.PREFIX == '/usr' and isdebianbased()):
+            templatedir_location = os.path.join(
+                '/usr/local',
+                file_location[0])
+        else:
+            templatedir_location = os.path.join(
+                distutils.sysconfig.PREFIX,
+                file_location[0])
+# Update conf file
+config = ConfigParser.ConfigParser()
+config.read(configfile_location)
+config.set('redfish-client', 'templates_path', templatedir_location)
+with open(configfile_location, 'w') as configfile:
+    config.write(configfile)
+
```

Comment:
Could you submit separately the renaming of config and inventory so I could already merge that, and then I want to resync with you wrt paths, as we shouldn't do it internally but externally.